UI: Fix updating static roles via role edit page on UI (#29498)
hashicorp/vault: changelog/29498.txt

```release-note:bug

> ui: Fixes 'cannot update static username' error when updating a static role's rotation period via the UI 

Reviewer: Great changelog message! This clearly explains the fix in a way easily understandable for users. Quite a few resources managed in the UI have static roles, can we mention that this is specifically for databases? (I also removed the "via UI" since the changelog is prefixed by `ui:`) Suggestions below, but word however you like 😄 
```suggestion
ui: Fixes 'cannot update static username' error when updating rotation period for a database static role
```

(This is a newer pattern we discussed recently in UI sync that we haven't really started yet)
```suggestion
ui/database: Fixes 'cannot update static username' error when updating static role's rotation period
```
